### PR TITLE
feat(blob): add createFolder method

### PR DESCRIPTION
**Pull request by vvo**
And remove variadic calls from main put methods, which will allow us to more
easily upgrade to newer TS versions.

Related:
- https://github.com/vercel/storage/pull/682
- https://github.com/vercel/storage/pull/667

### CODE REVIEW

File: packages/blob/src/create-folder.ts [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+  pathname: string;
+  /**
+   * URL of created folder. There is not content at this URL.
+   * Use this URL to delete the folder, just like you would delete a file.
+   */
+  url: string;
+}
+
+/**
+ * Creates a folder in your store. Vercel Blob has no real concept of folders, our file browser on Vercel.com displays folders based on the presence of trailing slashes in the pathname. Unless you are building a file browser system, you probably don't need to use this method.
```
Comment (luismeyer) [inline packages/blob/src/create-folder.ts:11]: should we warn here, that folders will appear as blobs when you call `list()`? https://github.com/vercel/storage/issues/640

File: test/next/src/app/vercel/blob/script.mts [hunk 2]
```diff
@@ -18,26 +18,26 @@ console.log();
...
-    axiosExample(),
-    gotExample(),
-    fetchExample(),
-    noExtensionExample(),
-    weirdCharactersExample(),
-    copyTextFile(),
-    listFolders(),
-    multipartNodeJsFileStream(),
-    fetchExampleMultipart(),
+    // textFileExample(),
```
Comment (luismeyer) [inline test/next/src/app/vercel/blob/script.mts]: this is for testing I guess

File: packages/blob/src/put.ts [hunk 3]
```diff
@@ -54,14 +43,14 @@ export function createPutMethod<TOptions extends PutCommandOptions>({
     const options = await createPutOptions({
       pathname,
       // when no body is required (for folder creations) options are the second argument
```
Comment (luismeyer) [inline packages/blob/src/put.ts]: this can be removed

**Review by luismeyer — APPROVED** [review]

File: test/next/src/app/vercel/blob/script.mts (hunk 2 above)
Comment (vvo) [reply test/next/src/app/vercel/blob/script.mts]: yes will remove

**Pull request merged**